## Healthcare ML Project (SPRINT 1)

### Import Required Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore') 

### Dataset Loading

In [2]:
df = pd.read_csv(r"E:\IRL\IRL_Projects\ML\1.HealthCare\Dataset\insurance.csv")
df.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [3]:
df = df.rename(columns={
    'age': 'Age',
    'sex': 'Gender',
    'bmi': 'BMI',
    'children': 'Childrens',
    'smoker': 'Smoker',
    'region': 'Region',
    'charges': 'Charges',
})

In [4]:
df

Age  Gender     BMI  Childrens Smoker     Region      Charges
0      19  female  27.900          0    yes  southwest  16884.92400
1      18    male  33.770          1     no  southeast   1725.55230
2      28    male  33.000          3     no  southeast   4449.46200
3      33    male  22.705          0     no  northwest  21984.47061
4      32    male  28.880          0     no  northwest   3866.85520
...   ...     ...     ...        ...    ...        ...          ...
1333   50    male  30.970          3     no  northwest  10600.54830
1334   18  female  31.920          0     no  northeast   2205.98080
1335   18  female  36.850          0     no  southeast   1629.83350
1336   21  female  25.800          0     no  southwest   2007.94500
1337   61  female  29.070          0    yes  northwest  29141.36030

[1338 rows x 7 columns]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        1338 non-null   int64  
 1   Gender     1338 non-null   object 
 2   BMI        1338 non-null   float64
 3   Childrens  1338 non-null   int64  
 4   Smoker     1338 non-null   object 
 5   Region     1338 non-null   object 
 6   Charges    1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


### Observation-
- There are 1338 Rows and 7 Columns in this dataset.

In [6]:
df.describe().T

count          mean           std        min         25%  \
Age        1338.0     39.207025     14.049960    18.0000    27.00000   
BMI        1338.0     30.663397      6.098187    15.9600    26.29625   
Childrens  1338.0      1.094918      1.205493     0.0000     0.00000   
Charges    1338.0  13270.422265  12110.011237  1121.8739  4740.28715   

                50%           75%          max  
Age          39.000     51.000000     64.00000  
BMI          30.400     34.693750     53.13000  
Childrens     1.000      2.000000      5.00000  
Charges    9382.033  16639.912515  63770.42801

### Checking Missing Values

In [7]:
df.isnull().sum()

Age          0
Gender       0
BMI          0
Childrens    0
Smoker       0
Region       0
Charges      0
dtype: int64

### Observation -
- There are no missing values in this dataset

### Checking Duplicate Values

In [8]:
df.duplicated().sum()

np.int64(1)

### Drop duplicates

In [9]:
df = df.drop_duplicates().reset_index(drop=True)

In [10]:
df.duplicated().sum()

np.int64(0)

In [11]:
df.shape

(1337, 7)

In [12]:
df

Age  Gender     BMI  Childrens Smoker     Region      Charges
0      19  female  27.900          0    yes  southwest  16884.92400
1      18    male  33.770          1     no  southeast   1725.55230
2      28    male  33.000          3     no  southeast   4449.46200
3      33    male  22.705          0     no  northwest  21984.47061
4      32    male  28.880          0     no  northwest   3866.85520
...   ...     ...     ...        ...    ...        ...          ...
1332   50    male  30.970          3     no  northwest  10600.54830
1333   18  female  31.920          0     no  northeast   2205.98080
1334   18  female  36.850          0     no  southeast   1629.83350
1335   21  female  25.800          0     no  southwest   2007.94500
1336   61  female  29.070          0    yes  northwest  29141.36030

[1337 rows x 7 columns]

In [13]:
df.to_csv('insurance-1.csv', index=False)